e2e_tests/tests/conftest.py

```diff
@@ -132,8 +132,7 @@ def pytest_itemcollected(item: Any) -> None:
         pytest.exit(f"{item.nodeid} is missing an integration test mark (any of {_INTEG_MARKERS})")
 
 
-@pytest.fixture(scope="session")
-def secrets(request: SubRequest) -> Dict[str, str]:
+def s3_secrets(request: SubRequest) -> Dict[str, str]:
```

Comment:
question: why break this into a separate function from secrets?

Reply:
By breaking the function into two we can allow the standard `secrets` function to `pytest.skip` if the secrets are not available, where we want `checkpoint_storage_config` to require the `s3_secrets` if configured for s3, where an error would be raised if the secrets are not available.